lib/java/com/google/android/material/lists/MaterialListAdapter.java

```diff
@@ -0,0 +1,73 @@
+package com.google.android.material.lists;
+
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.android.material.lists.listdatatypes.ArrayListData;
+import com.google.android.material.lists.listdatatypes.CustomListData;
+import com.google.android.material.lists.listdatatypes.ListListData;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+/**
+ * An adapter that uses {@link MaterialViewHolder}s and binds data to them using {@link MaterialListBinder}
+ */
+public class MaterialListAdapter<T, E extends MaterialViewHolder> extends RecyclerView.Adapter<E> {
+
+  private CustomListData<T> customListData;
+  private MaterialListBinder<T, E> materialListBinder;
+  private Class<E> viewHolderClass;
+
+  public MaterialListAdapter(MaterialListBinder<T, E> materialListBinder, Class<E> viewHolderClass, T[] data) {
+    this.materialListBinder = materialListBinder;
+    this.viewHolderClass = viewHolderClass;
+    customListData = new ArrayListData<>(data);
+  }
+
+  public MaterialListAdapter(MaterialListBinder<T, E> materialListBinder, Class<E> viewHolderClass, List<T> data) {
+    this.materialListBinder = materialListBinder;
+    this.viewHolderClass = viewHolderClass;
+    customListData = new ListListData<>(data);
+  }
+
+  public MaterialListAdapter(MaterialListBinder<T, E> materialListBinder, Class<E> viewHolderClass, CustomListData<T> data) {
+    this.materialListBinder = materialListBinder;
+    this.viewHolderClass = viewHolderClass;
+    customListData = data;
+  }
+
+  @NonNull
+  @Override
+  public E onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+
+    try {
+      return viewHolderClass.getConstructor(ViewGroup.class).newInstance(parent);
+    } catch (IllegalAccessException e) {
+      e.printStackTrace();
+    } catch (InstantiationException e) {
+      e.printStackTrace();
+    } catch (InvocationTargetException e) {
+      e.printStackTrace();
+    } catch (NoSuchMethodException e) {
+      e.initCause(
+          new NoSuchMethodException(viewHolderClass.getName() +
+              " needs a constructor with only a ViewGroup parameter"));
+      e.printStackTrace();
+    }
```

Comment:
Could you explain why you're printing the stack trace and not actually handling the exceptions?

Reply:
Other than printing why the error occurred what else can I do to "handle the exceptions" ?

Just for clarification, the IllegalAccessException, InstantiationException and InvocationTargetException cannot happen under normal circumstances. However, the NoSuchMethodException can happen so I added a message about why it happened.